File: src/main/java/org/cloudfoundry/credhub/request/RegenerateRequest.java
package org.cloudfoundry.credhub.request;

import javax.validation.constraints.NotNull;

import com.fasterxml.jackson.annotation.JsonAutoDetect;
import org.apache.commons.lang3.StringUtils;

@JsonAutoDetect
@SuppressWarnings("unused")
public class RegenerateRequest {

  @NotNull(message = "error.missing_name")
  private String name;

  public RegenerateRequest() {
    super();
    /* this needs to be there for jackson to be happy */
  }

  public RegenerateRequest(final String name) {
    super();
    this.name = name;
  }

  public String getName() {
    return name;
  }

  public void setName(final String name) {
    this.name = StringUtils.prependIfMissing(name, "/");
  }
}

File: components/generate/src/main/kotlin/org/cloudfoundry/credhub/requests/RegenerateRequest.kt
package org.cloudfoundry.credhub.requests

import com.fasterxml.jackson.annotation.JsonAutoDetect
import com.fasterxml.jackson.databind.JsonNode
import jakarta.validation.constraints.NotNull
import org.cloudfoundry.credhub.ErrorMessages

@JsonAutoDetect
class RegenerateRequest {
    @NotNull(message = ErrorMessages.MISSING_NAME)
    private lateinit var name: String

    private var metadata: JsonNode? = null

    constructor() {
        // this needs to be there for jackson to be happy
    }

    fun getName(): String = name

    fun getMetadata(): JsonNode? = metadata
}
